src/polynomial/mod.rs

```diff
@@ -91,7 +128,7 @@ impl<F: PrimeField> Expression<F> {
         }
     }
 
-    fn _expand(&self, index_to_poly: &Vec<(i32, usize)>) -> MultiPolynomial<F> {
+    fn _expand(&self, index_to_poly: &Vec<(i32, usize, usize)>) -> MultiPolynomial<F> {
```

Comment:
`&Vec` - anti-pattern, use `&[(i32, usize, usize)]`

Reply:
Such use of types in general looks incomprehensible, at least explain in rustdoc what this magic trait of three indexes is

Reply:
will use `&[(i32, usize, usize)]` instead. 
As for the type explanation, it has been explained in the definition of `index_to_poly` (see `pub struct Monomial`)

Reply:
> `&Vec` - anti-pattern, use `&[(i32, usize, usize)]`

If it's anti-pattern, I will change it. I have one question: if we use slice here, then inside the function we use something like:
`vec![Monomial::new(index_to_poly.to_vec(), F::ONE, exponents)]`

Is this way more efficient or they are the same?

Reply:
> For the type explanation, it has been explained in the definition of index_to_poly (see pub struct Monomial)

Well, either a normal struct should be created that represents these indexes (I am in favor of this option), or there should be an explanation at each place of use, or in the worst case a reference to the explanation in one place

> Is this way more efficient or they are the same?

Completely the same. In both cases there will be one call of mem copy.